From 1585b73b5760d1c6688c620f2e599ef8cb46ffac Mon Sep 17 00:00:00 2001
From: Kevin Strasser <kevin.strasser@linux.intel.com>
Date: Wed, 12 Sep 2012 20:39:29 -0700
Subject: [PATCH] add UYVY for caspa support

add the UYVY format that is provided by ISP to the list of formats
tried for connection

Signed-off-by: Kevin Strasser <kevin.strasser@linux.intel.com>
---
 sys/v4l2/gstv4l2object.c |   15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/sys/v4l2/gstv4l2object.c b/sys/v4l2/gstv4l2object.c
index 81d1cb7..9c61b96 100644
--- a/sys/v4l2/gstv4l2object.c
+++ b/sys/v4l2/gstv4l2object.c
@@ -1098,6 +1098,21 @@ gst_v4l2_object_fill_format_list (GstV4l2Object * v4l2object)
         (GCompareFunc) format_cmp_func);
   }
 
+/* Add UYVY format to the list of enumerations */
+format = g_new0 (struct v4l2_fmtdesc, 1);
+format->type = (enum v4l2_buf_type)-1;
+format->index = n;
+format->flags = 0;
+format->pixelformat = V4L2_PIX_FMT_UYVY;
+v4l2object->formats = g_slist_prepend (v4l2object->formats, format);
+GST_INFO_OBJECT (v4l2object->element, "index: %u", format->index);
+GST_INFO_OBJECT (v4l2object->element, "type: %d", format->type);
+GST_INFO_OBJECT (v4l2object->element, "flags: %08x", format->flags);
+GST_INFO_OBJECT (v4l2object->element, "description: '%s'",
+       format->description);
+GST_INFO_OBJECT (v4l2object->element, "pixelformat: %" GST_FOURCC_FORMAT,
+GST_FOURCC_ARGS (format->pixelformat));
+
 #ifndef GST_DISABLE_GST_DEBUG
   {
     GSList *l;
-- 
1.7.9.5

